prow/bugzilla/client.go

```diff
@@ -89,6 +91,43 @@ func NewClient(getAPIKey func() []byte, endpoint string) Client {
 	}
 }
 
+// newBugDetailsCache is a constructor for bugDetailsCache
+func newBugDetailsCache() *bugDetailsCache {
+	return &bugDetailsCache{cache: map[int]Bug{}}
+}
+
+// bugDetailsCache holds the already retrieved bug details
+type bugDetailsCache struct {
+	cache map[int]Bug
+	lock  sync.Mutex
+}
+
+// get retrieves bug details from the cache and is thread safe
+func (bd *bugDetailsCache) get(key int) (bug Bug, exists bool) {
+	bd.lock.Lock()
+	defer bd.lock.Unlock()
+	entry, ok := bd.cache[key]
+	return entry, ok
+}
+
+// set stores the bug details in the cache and is thread safe
+func (bd *bugDetailsCache) set(key int, value Bug) {
+	bd.lock.Lock()
+	defer bd.lock.Unlock()
+	bd.cache[key] = value
+}
+
+// list returns a slice of all bugs in the cache
+func (bd *bugDetailsCache) list() []Bug {
+	bd.lock.Lock()
+	defer bd.lock.Unlock()
+	result := []Bug{}
```

Comment:
Not sure how large these should be in practice but we can allocate a large enough slice right away and possibly avoid reallocation.

Reply:
I have no way to know how large the clones set would be, but I guess 10 would be a reasonable number beyond which most would not go. I can make that change then

Reply:
`len(cache.cache)` is the max size of the set.